File: src/cli/tests/fixtures/doc/doc005_undocumented_param.rs
//! Rule: DOC005 - every parameter must be mentioned in the `# Arguments` section.
//!
//! When a `# Arguments` section exists, each parameter name should appear
//! somewhere in it. Parameters that are not mentioned are flagged.
//!
//! Expected diagnostics:
//! - DOC005 on `pub fn build` (param `fmt` is not mentioned)
//!
//! Not flagged (should pass):
//! - `pub fn render` (both `text` and `width` are documented)

/// Builds output from a name and format.
///
/// # Arguments
///
/// `name` - the name to use.
pub fn build(name: &str, fmt: &str) -> String {
    format!("{name} {fmt}")
}

/// Renders text at a given width.
///
/// # Arguments
///
/// `text` - the text to render.
/// `width` - the target width.
pub fn render(text: &str, width: u32) -> String {
    format!("{text} {width}")
}

File: src/cli/tests/fixtures/doc/csharp/doc005_undocumented_param.cs
// DOC005 for C#: <param> tags must name every declared parameter.
namespace Fixtures;

/// <summary>Builds things.</summary>
public class Builder
{
    /// <summary>Builds with one parameter undocumented.</summary>
    /// <param name="name">The name.</param>
    public void Build(string name, string format) { }

    /// <summary>Builds with every parameter documented.</summary>
    /// <param name="name">The name.</param>
    /// <param name="format">The format.</param>
    public void Built(string name, string format) { }
}
